src/main/java/io/jenkins/plugins/opentelemetry/OtelUtils.java

```diff
@@ -40,24 +40,72 @@
 
     @Nonnull
     public static String getProjectType(Run run) {
-        if (run == null) {
-            return "unknown";
-        }
-        if (run instanceof FreeStyleBuild) {
+        if (isFreestyle(run)) {
             return "freestyle";
         }
-        if (run instanceof WorkflowRun) {
-            if (run.getParent().getParent() instanceof WorkflowMultiBranchProject) {
-                return "multibranch";
-            } else {
-                return "workflow";
-            }
+        if (isMultibranch(run)) {
+            return "multibranch";
+        }
+        if (isWorkflow(run)) {
+            return "workflow";
         }
         // TODO: support for
         // https://github.com/jenkinsci/matrix-project-plugin/blob/master/src/main/java/hudson/matrix/MatrixBuild.java#L70
         return "unknown";
     }
 
+    @Nonnull
+    public static String getMultibranchType(Run run) {
+        if (isMultibranch(run)) {
+            if (isMultibranchPullRequest(run)) {
+                return "pr";
+            }
+            if (isMultibranchBranch(run)) {
+                return "branch";
+            }
+            // TODO: discover tag type.
+        }
+        return "unknown";
+    }
+
+    public static boolean isMultibranchPullRequest(Run run) {
+        if (isMultibranch(run)) {
+            return run.getParent().getName().startsWith("PR-");
```

Comment:
We should use:
```
SCMHead head = SCMHead.HeadByItem.findHead(run.getParent());
return (head instanceof ChangeRequestSCMHead);
```

"PR-" woks for GitHub and BitBucket but not for GitLab.

See https://github.com/jenkinsci/opentelemetry-plugin/blob/4cbcd2abc48fe78e4ed7652cd6b15f4311ef363f/src/main/java/io/jenkins/plugins/opentelemetry/job/SpanNamingStrategy.java#L43

Reply:
See https://github.com/jenkinsci/opentelemetry-plugin/pull/52/commits/9267d226e1016929d25e125ce108e6f677842dd0